File: apps/question-service/handlers/delete.go
package handlers

import (
	"fmt"
	"net/http"

	"github.com/go-chi/chi/v5"
	"google.golang.org/grpc/codes"
	"google.golang.org/grpc/status"
)

func (s *Service) DeleteQuestion(w http.ResponseWriter, r *http.Request) {
	ctx := r.Context()

	// Parse request
	docRefID := chi.URLParam(r, "docRefID")

	// Reference document
	docRef := s.Client.Collection("questions").Doc(docRefID)

	// Validation
	// Check if exists
	_, err := docRef.Get(ctx)
	if err != nil {
		if status.Code(err) == codes.NotFound {
			http.Error(w, "Question not found", http.StatusNotFound)
			return
		}
		http.Error(w, "Error fetching question", http.StatusInternalServerError)
		return
	}

	// Update database
	_, err = docRef.Delete(ctx)
	if err != nil {
		http.Error(w, "Error deleting question", http.StatusInternalServerError)
		return
	}

	fmt.Fprintf(w, "Question with ID %s deleted successfully", docRefID)
}

File: question-service/controller/delete.js
import Question from '../model/Question.js';

export const deleteQuestion = async (req, res) => {
    try {
      const id = req.params.id;
      const filter = {
        id: id
      }
      const question = await Question.findOneAndDelete(filter);
      return res.status(200).json(question);;
    } catch (err) {
      return res.status(500).json({ 
        message: `Error when deleting question with id ${id}!`,
        error: err
      });
    }
};